added new branch

anusharm15 opened the pull request:
Hi Ankur,

As you said , i created an empty branch. Please review the code in master branch and you can check live application which deployed here : https://demoapp-8fd72.firebaseapp.com/ .

Following things are not implemented as per the requirement document :
=> Save for Later
=> See product details link
=> ENTER PROMOTION CODE and Apply coupon

Thanks

ankurkakroo2 commented:
Before going into specifics, here is a high level observation:

You should think on the grounds of a scalability. Atomic deisgn (http://bradfrost.com/blog/post/atomic-web-design/) can be be good start.

Right now everything resides in one single file.
Let me know if you need any discussion around this.

anusharm15 commented:
Thanks for the valuable feedback Ankur.

Actually I thought there is not much functionality so i only used 3 files script.js (here i am waiting to DOM load and import other files), service.js( here i am handling ajax call and returning promis) and render.js. I put all logic and dynamic html rendering inside "render.js". As per your suggestion, I will make it scalable devide "render.js" into controller and view part.  Any other suggestions if you can give  would be helpful so i will look into that also.

anusharm15 commented:
Hi Ankur,

I have done some modifications in the structure  and changed some code also. Also removed the react dependency from package.json. Can you please check.

Note : please ignore the bootstrap.min.css i have added in index.html. I am not using bootstrap. By mistake it is added and now when i try to remove it , it showing some ui issues. So, for now i just keep it there.
